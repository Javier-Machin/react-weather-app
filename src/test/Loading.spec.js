import React from 'react';
import { shallow } from 'enzyme';
import LoadingMessage from '../components/Loading';

describe('LoadingMessage', () => {
  it('should render correctly', () => {
    const component = shallow(<LoadingMessage />);
    expect(component).toMatchSnapshot();
  });

  it('should display a loading message when passed loading true prop', () => {
    const component = shallow(<LoadingMessage loading />);
    expect(component.find('div.loading-true')).toHaveLength(1);
  });

  it('should not display a loading message when passed loading false prop', () => {
    const component = shallow(<LoadingMessage loading={false} />);
    expect(component.find('div.loading-true')).toHaveLength(0);
  });
});
