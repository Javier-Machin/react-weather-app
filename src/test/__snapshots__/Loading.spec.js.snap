// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingMessage should render correctly 1`] = `
<div
  className="loading-message"
>
  Loading
</div>
`;
